IPA Test Scripts

dtapiacl opened the pull request:
Python snippet test

dtapiacl committed:
ipa-account-script

dtapiacl committed:
Update to script

dtapiacl committed:
add_flag_param

dtapiacl committed:
ipa-py-script

dtapiacl committed:
Update to code format

jhoblitt commented on pythontest.py:
```diff
@@ -0,0 +1,20 @@
+#Import Modules
```
A shebang line is usually a good idea.

jhoblitt commented on pythontest.py:
```diff
@@ -0,0 +1,20 @@
+for d in data:
+    subprocess.run(["ipa", "user-del", "--preserve", "--dry-run", d['uid'][0]])
+    
+print(d)
```
Python is sensitive to whitespace.  This statement is probably a logic error in that `print()` will only be invoked once and when it is, the variable `d` will be out of scope.

jhoblitt commented on pythontest.py:
```diff
@@ -0,0 +1,20 @@
+import json 
+import subprocess
+
+#Open json file and parse its contents inside the data variable using json.load method
```
I find it much easier to ready comments if there is a space between the `#` and the first word.

jhoblitt commented on pythontest.py:
```diff
@@ -0,0 +1,20 @@
+
+
+
+
```
This trailing whitespace seems unnecessary.

jhoblitt commented on user.sh:
```diff
@@ -0,0 +1,10 @@
+    echo "User $user was deleted"
+done
+echo "All users have been deleted"
+
```
This trailing whitespace seems unnecessary.

jhoblitt commented on pythontest.py:
```diff
@@ -0,0 +1,20 @@
+with open("disabled_users.json", "r") as json_file:
+    data = json.load(json_file)
+
+#For loop to iterate the contents of data and delete the users with key 'uid' in dry-run mode      
```
Needs a space after the `#`.

jhoblitt reviewed (changes requested):
- " It 4484" is not needed in the title.  Jira will only match on the exact ticket slug and the slug is already part of the branch name.
- As with commit messages, the PR title should be imperative. See: https://cbea.ms/git-commit/
- Commit b05617ce3a7aa99381075da34727ee53e078ea15 should have an imperative commit message.
- The commit message for 2683ef308fc23630a89f49f02faa8f755bda6a34 doesn't state what has changed.
- The commit message for 91d577ce96a759f30074c2cf27ca3e26836304a0 seems to have an extra `_` in it. There should be a space after "add".
- Commit message 33d9b609901b0cbb4628ad486552536e82567fa0 should have an imperative commit message.
- The commit message for c23dc6c2d52b26371b9e46aa7c0144bb6764490a should have a space after "add" instead of a "-".

dtapiacl committed:
removed trailing white space

dtapiacl commented:
> * " It 4484" is not needed in the title.  Jira will only match on the exact ticket slug and the slug is already part of the branch name.
> 
>     * As with commit messages, the PR title should be imperative. See: https://cbea.ms/git-commit/
> 
>     * Commit [b05617c](https://github.com/lsst-it/ipa-scripts/commit/b05617ce3a7aa99381075da34727ee53e078ea15) should have an imperative commit message.
> 
>     * The commit message for [2683ef3](https://github.com/lsst-it/ipa-scripts/commit/2683ef308fc23630a89f49f02faa8f755bda6a34) doesn't state what has changed.
> 
>     * The commit message for [91d577c](https://github.com/lsst-it/ipa-scripts/commit/91d577ce96a759f30074c2cf27ca3e26836304a0) seems to have an extra `_` in it. There should be a space after "add".
> 
>     * Commit message [33d9b60](https://github.com/lsst-it/ipa-scripts/commit/33d9b609901b0cbb4628ad486552536e82567fa0) should have an imperative commit message.
> 
>     * The commit message for [c23dc6c](https://github.com/lsst-it/ipa-scripts/commit/c23dc6c2d52b26371b9e46aa7c0144bb6764490a) should have a space after "add" instead of a "-".

Thanks, @jhoblitt, for the comments, I fixed most of my issues, but I could not fix and update my commit history for the bash script as I did not find those commits in my git log.  I assume this happened because I deleted the script from the remote repository which was probably a bad idea.  I was able to update my python3 commit history without issues.

dtapiacl committed:
(ipa_scripts) add license file

dtapiacl committed:
Merge pull request #3 from lsst-it/IT-5292_add_license_ipa_scripts

(ipa_scripts) add license file

jhoblitt reviewed (approved):
Please rebase on `main` to remove the merge commit.
